text/0000-volta-updates.md

```diff
@@ -34,7 +34,16 @@ For Volta updates, there are three distinct sets of changes that need to be cons
 
 The large majority of updates to Volta involve _only_ updating the binaries, without requiring updates to the Data Directory or Profile.
 
-For Managed users, updating the binaries will be handled by the team managing the distribution, making sure that they are always on the latest binaries.
+### Managed Users
+
+For Managed users, updating the binaries will be handled by the team managing the distribution, making sure that they are always on the latest binaries. Specifically, we currently provide 2 binaries:
```

Comment:
Does this mean that adding a new binary will be a "breaking change" to managed users? How do we properly communicate that a new required binary is present (or that we have removed one, if we ever decide to do that)?

I suspect we need some official metadata format that will be parsed by teams doing the managed distribution, otherwise that would always be a completely manual (and therefore error prone) process.

Thoughts?

Reply:
I imagine we'll be distributing these in some sort of package (i.e. a tarball), so would "the contents of this archive" count as a metadata format? That's how we currently distribute the installed files for all of the Unix OS's: A tarball with all the necessary files that gets unpacked by the installer.

We could also provide some sort of "Manifest" file that lists all the binaries that are needed in order for the update to work, though I'm wary of trying to create a new format, that then will need to be translated manually into whatever format is used by the final users. Perhaps a simple list file, one file name per line, that can be published as part of the release, alongside the tarball?

Reply:
> would "the contents of this archive" count as a metadata format?

Yes, that seems totally fine (something like "the tarball provided is unpacked directly onto the users `$PATH`). My only concern here is that (as far as I understand the system) we are **not** saying the "full tarball" is unpacked somewhere on the `$PATH` (we are saying "these 2 binaries" have to be on `$PATH`). I admit that I haven't actually dug into the contents of that tarball so maybe I'm just missing something silly here.

> Perhaps a simple list file, one file name per line, that can be published as part of the release, alongside the tarball?

Ya, this seems good to me also, though I think it would be perfectly fine if this `BinaryFiles.txt` file (or whatever we call it 😛) is actually **inside** the root of the tarball. That makes it pretty easy to bundle everything together, and it makes it a bit clearer what the filenames are "relative to".

Reply:
Currently the tarball includes the 2 binary files (`volta` and `shim`) as well as the two shell-specific function wrappers (`load.sh` and `load.fish`). With the shift to basing everything on the binaries themselves, the latter two would be going away, so the tarball would actually only include the necessary binary files.

That said, it's straightforward enough to also create / include a manifest file of some kind inside the tarball, so I'm open to either solution.

Reply:
> Currently the tarball includes the 2 binary files (volta and shim) as well as the two shell-specific function wrappers (load.sh and load.fish).

I see! I'm sorry, I didn't realize that was all that the current tarball includes. If we think that will continue to be the case, we can definitely say "all of the tarball contents get extracted to somewhere on the users $PATH", but I _suspect_ that we might also want to include some other files eventually (e.g. `README.md`, `LICENSE`, etc) so if the manifest route isn't _too_ bad maybe we should still do it?

Reply:
Makes sense to me, and it shouldn't be bad at all, since it will change very rarely. We can create a `volta.manifest` file and include it in the tarball, then provide documentation that indicates what needs to be included in an "install" are only the files listed in the manifest.